### PR TITLE
Add gallery example showing usage of polygon objects from a geopandas.GeoDataFrame (choropleth map)

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Here's the second gallery example showcasing how to plot objects stored in a geopandas.GeoDataFrame, now for polygons. 

Related to #1374, #1474, #2786, #2798.  

**Preview**: https://pygmt-dev--2796.org.readthedocs.build/en/2796/gallery/maps/choropleth_map.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/gallery/lines/polygons.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+The :meth:`pygmt.Figure.plot` method allows us to plot geographical data such
+as polygons which are stored in a :class:`geopandas.GeoDataFrame` object. Use
+:func:`geopandas.read_file` to load data from any supported OGR format such as
+a shapefile (.shp), GeoJSON (.geojson), geopackage (.gpkg), etc. You can also
+use a full URL pointing to your desired data source. Then, pass the
+:class:`geopandas.GeoDataFrame` as an argument to the ``data`` parameter in
+:meth:`pygmt.Figure.plot`, and style the geometry using the ``pen`` parameter.
+To fill the polygons based on a corresponding column you need to set
+``fill="+z"```as well as select the appropriate column using the ``aspatial``
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/polygons.py:13]:
```suggestion
``fill="+z"`` as well as select the appropriate column using the ``aspatial``
```

File: examples/gallery/lines/polygons.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+"""
+Filled polygons
+===============
+
+The :meth:`pygmt.Figure.plot` method allows us to plot geographical data such
+as polygons which are stored in a :class:`geopandas.GeoDataFrame` object. Use
+:func:`geopandas.read_file` to load data from any supported OGR format such as
+a shapefile (.shp), GeoJSON (.geojson), geopackage (.gpkg), etc. You can also
+use a full URL pointing to your desired data source. Then, pass the
+:class:`geopandas.GeoDataFrame` as an argument to the ``data`` parameter in
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/polygons.py:10]:
```suggestion
:class:`geopandas.GeoDataFrame` as an argument to the ``data`` parameter of
```

File: examples/gallery/lines/polygons.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    water="lightblue",
+    land="gray70",
+    shorelines="1/1p,gray70",
+)
+
+# The dataset contains different parameters, here we select
+# the "population" column to plot.
+
+# First, we define the colormap to fill the polygons based on
+# only the "population" column.
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/polygons.py:43]:
```suggestion
# First, we define the colormap to fill the polygons based on
# the "population" column.
```

File: examples/gallery/lines/polygons.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+fig = pygmt.Figure()
+
+fig.coast(
+    region=[bounds[0] - edge, bounds[2] + edge, bounds[1] - edge, bounds[3] + edge],
+    projection="M10c",
+    frame=["af", "+tPopulation of Chicago"],
+    water="lightblue",
+    land="gray70",
+    shorelines="1/1p,gray70",
+)
```
Comment (seisman) [inline examples/gallery/lines/polygons.py:37]: What about removing the water/land/shorelines and also the frame:
```suggestion
fig.basemap(
    region=[bounds[0] - edge, bounds[2] + edge, bounds[1] - edge, bounds[3] + edge],
    projection="M10c",
    frame="+tPopulation of Chicago",
)
```

I feel the plot below is more beautiful:
map

File: examples/gallery/lines/polygons.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+# The dataset contains different parameters, here we select
+# the "population" column to plot.
+
+# First, we define the colormap to fill the polygons based on
+# only the "population" column.
+pygmt.makecpt(
+    cmap="acton",
+    series=[np.min(gdf["population"]), np.max(gdf["population"]), 10],
```
Comment (seisman) [inline examples/gallery/lines/polygons.py:46]:
```suggestion
    series=[gdf["population"].min(), gdf["population"].max(), 10],
```

File: examples/gallery/lines/polygons.py (hunk 4 above)
Comment (michaelgrund) [reply examples/gallery/lines/polygons.py]: Agree, the simpler the better!

File: examples/gallery/lines/polygons.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
+"""
+Filled polygons
```
Comment (seisman) [inline examples/gallery/lines/polygons.py:2]: Does it make sense to change the title to "Choropleth Map"?

File: examples/gallery/lines/polygons.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+gdf = gpd.read_file("https://geodacenter.github.io/data-and-lab//data/airbnb.zip")
+# Automatically get min/max coordinates of polygon set
+bounds = gdf.total_bounds
+# Define an edge in degrees to add in each direction to the min/max coordinates
+edge = 0.02
+
+fig = pygmt.Figure()
+
+fig.basemap(
+    region=[bounds[0] - edge, bounds[2] + edge, bounds[1] - edge, bounds[3] + edge],
```
Comment (seisman) [inline examples/gallery/lines/polygons.py:30]:
```suggestion

fig = pygmt.Figure()

fig.basemap(
    # Automatically get min/max coordinates of polygon set
    region=gdf.total_bounds[[0, 2, 1, 3]],
```

File: examples/gallery/lines/polygons.py (hunk 6 above)
Comment (yvonnefroehlich) [reply examples/gallery/lines/polygons.py:2]: Then I suggest to also rename the file itself.

Comment (michaelgrund) [reply examples/gallery/lines/polygons.py]: Done!

File: examples/gallery/lines/choropleth_map.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+To fill the polygons based on a corresponding column you need to set
+``fill="+z"`` as well as select the appropriate column using the ``aspatial``
+parameter as shown in the example below.
+"""
+
+import geopandas as gpd
+import pygmt
+
+# Read polygon data using geopandas
+gdf = gpd.read_file("https://geodacenter.github.io/data-and-lab//data/airbnb.zip")
```
Comment (seisman) [inline examples/gallery/lines/choropleth_map.py:21]:
```suggestion
gdf = gpd.read_file("https://geodacenter.github.io/data-and-lab/data/airbnb.zip")
```

File: examples/gallery/lines/choropleth_map.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+fig = pygmt.Figure()
+
+fig.basemap(
+    # Automatically get min/max coordinates of polygon set
```
Comment (seisman) [inline examples/gallery/lines/choropleth_map.py:26]: Is it OK to remove this comment?
```suggestion
    # Automatically get min/max coordinates of polygon set
```

File: examples/gallery/lines/choropleth_map.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    region=gdf.total_bounds[[0, 2, 1, 3]],
+    projection="M10c",
+    frame="+tPopulation of Chicago",
+)
+
+# The dataset contains different parameters, here we select
```
Comment (seisman) [inline examples/gallery/lines/choropleth_map.py:32]: I feel `parameters` is not suitable here, but I'm also unsure if `attributes` is the correct word.
```suggestion
# The dataset contains different attributes, here we select
```

File: examples/gallery/lines/choropleth_map.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    pen="0.3p,gray10",
+    close=True,
+    fill="+z",
+    cmap=True,
+    aspatial="Z=population",
+)
+
+
+# Add colorbar legend
+fig.colorbar(frame="x+lPopulation")
```
Comment (seisman) [inline examples/gallery/lines/choropleth_map.py:58]: Looking at the image at https://pygmt-dev--2796.org.readthedocs.build/en/2796/gallery/lines/choropleth_map.html, there are a lot blank spaces. Maybe put a vertical colorbar at the lower-left corner?

**Comment by yvonnefroehlich** [comment]
Maybe it makes more sense to place this example in the "maps" folder instead of the "lines" folder? Probably it is currently stored in the "lines" folder, as there is also the example for the line geometries. But this example is not really about lines.

File: examples/gallery/lines/choropleth_map.py [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    data=gdf[["population", "geometry"]],
+    pen="0.3p,gray10",
+    close=True,
+    fill="+z",
+    cmap=True,
+    aspatial="Z=population",
+)
+
+
+# Add colorbar legend
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/choropleth_map.py:57]:
```suggestion
    aspatial="Z=population",
)

# Add colorbar legend
```

File: examples/gallery/lines/choropleth_map.py [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+pygmt.makecpt(
+    cmap="acton",
+    series=[gdf["population"].min(), gdf["population"].max(), 10],
+    continuous=True,
+    reverse=True,
+)
+
+# Next, we plot the polygons and fill them using the defined
+# colormap. The target column is defined by the aspatial
+# parameter.
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/choropleth_map.py:46]:
```suggestion
# Next, we plot the polygons and fill them using the defined colormap.
# The target column is defined by the aspatial parameter.
```

File: examples/gallery/lines/choropleth_map.py (hunk 10 above)
Comment (yvonnefroehlich) [reply examples/gallery/lines/choropleth_map.py:32]: I am also wondering whether "parameters" is here to optimal word. I was thinking about using "quantities":
```suggestion
# The dataset contains different quantities, here we select
```

**Comment by michaelgrund** [comment]
> Maybe it makes more sense to place this example in the "maps" folder instead of the "lines" folder? Probably it is currently stored in the "lines" folder, as there is also the example for the line geometries. But this example is not really about lines.

Done!

File: examples/gallery/lines/choropleth_map.py (hunk 11 above)
Comment (michaelgrund) [reply examples/gallery/lines/choropleth_map.py]: Made an adjustment!

**Comment by michaelgrund** [comment]
Will set this PR on draft until #2798 is merged.

**Comment by seisman** [comment]
> Will set this PR on draft until #2798 is merged.

@michaelgrund I don't think I have time finishing PR #2798 in the next few weeks or one month. So it's up to you if you want (1) wait until PR #2798 is merged; or (2) merge this PR first, then simplify it in the future after #2798 is merged.

**Comment by michaelgrund** [comment]
> > Will set this PR on draft until #2798 is merged.
> 
> @michaelgrund I don't think I have time finishing PR #2798 in the next few weeks or one month. So it's up to you if you want (1) wait until PR #2798 is merged; or (2) merge this PR first, then simplify it in the future after #2798 is merged.

Then let's go with option (2) and simplify it later.

File: examples/gallery/maps/choropleth_map.py [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+    series=[gdf["population"].min(), gdf["population"].max(), 10],
+    continuous=True,
+    reverse=True,
+)
+
+# Next, we plot the polygons and fill them using the defined
+# colormap. The target column is defined by the aspatial
+# parameter.
+fig.plot(
+    data=gdf[["population", "geometry"]],
```
Comment (seisman) [inline examples/gallery/maps/choropleth_map.py:47]: Does this also work?
```suggestion
    data=gdf,
```

File: examples/gallery/maps/choropleth_map.py [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+import pygmt
+
+# Read polygon data using geopandas
+gdf = gpd.read_file("https://geodacenter.github.io/data-and-lab/data/airbnb.zip")
+
+fig = pygmt.Figure()
+
+fig.basemap(
+    region=gdf.total_bounds[[0, 2, 1, 3]],
+    projection="M10c",
```
Comment (seisman) [inline examples/gallery/maps/choropleth_map.py:27]: Maybe we should change `10c` to a smaller size like `6c`. The current figure is too a little too tall.

File: examples/gallery/maps/choropleth_map.py (hunk 14 above)
Comment (yvonnefroehlich) [reply examples/gallery/maps/choropleth_map.py:47]: Yes, for me it works to pass the complete GeoDataFrame.

File: examples/gallery/maps/choropleth_map.py [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+    reverse=True,
+)
+
+# Next, we plot the polygons and fill them using the defined
+# colormap. The target column is defined by the aspatial
+# parameter.
+fig.plot(
+    data=gdf,
+    pen="0.3p,gray10",
+    close=True,
```
Comment (yvonnefroehlich) [inline examples/gallery/maps/choropleth_map.py:49]: I feel we should explain that the `close` parameter is used to force closed polygons; maybe already in the introduction text at lines.

File: examples/gallery/maps/choropleth_map.py [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+    fill="+z",
+    cmap=True,
+    aspatial="Z=population",
+)
+
+
+# Add colorbar legend
+fig.colorbar(frame="x+lPopulation", position="jML+w6c/0.5c")
```
Comment (yvonnefroehlich) [inline examples/gallery/maps/choropleth_map.py:57]: With changing the size of the map, we probably have to adjust the colorbar :slightly_smiling_face:.

File: examples/gallery/maps/choropleth_map.py (hunk 16 above)
Comment (seisman) [reply examples/gallery/maps/choropleth_map.py:49]: Is `close` required?

Comment (yvonnefroehlich) [reply examples/gallery/maps/choropleth_map.py:49]: This is something I am not 100 % sure about. I orientated myself at the EGU short course example for a choropleth map (https://www.generic-mapping-tools.org/egu22pygmt/ecosystem.html#plotting-geospatial-vector-data-with-geopandas-and-pygmt). And there, the `close` parameter is set to `True` to force closed polygons. I already tried it without using this, and the figure looked correct to me.

Comment (seisman) [reply examples/gallery/maps/choropleth_map.py:49]: When plotting polygons without fills, `close` is required to connect the first and last points of polygons. 

When filling polygons, the polygons are always "closed" by default.

Comment (michaelgrund) [reply examples/gallery/maps/choropleth_map.py:49]: Then I would suggest to remove it in the example.

Comment (yvonnefroehlich) [reply examples/gallery/maps/choropleth_map.py:49]: Hm. Based on this, it seems to me, we do not need to explicitly set the `close` parameter to `True` when creating a choropleth map from polygons provided via a GeoDataFrame.

I just tried to plot the polygons of this GeoDataFrame without any fill. Also this works for me correctly without using `close=True`:
```python
import geopandas as gpd
import pygmt

gdf = gpd.read_file("https://geodacenter.github.io/data-and-lab/data/airbnb.zip")
fig = pygmt.Figure()

fig.basemap(region=gdf.total_bounds[[0, 2, 1, 3]], projection="M6c", frame="+t ")
fig.plot(data=gdf)

fig.show()
```
_Output figure_
geodataframe_withoutfill

File: examples/gallery/maps/choropleth_map.py (hunk 17 above)
Comment (michaelgrund) [reply examples/gallery/maps/choropleth_map.py]: Done!

**Review by seisman — APPROVED** [review]

File: examples/gallery/maps/choropleth_map.py (hunk 16 above)
Comment (seisman) [reply examples/gallery/maps/choropleth_map.py]: Here is an example showing how `close` works with/without fill.
```
import pygmt
fig = pygmt.Figure()
data = [[1, 1], [1, 5], [5, 5], [5, 1]]
fig.plot(data=data, projection="X10c/10c", frame=True, region=[0, 10, 0, 10], pen="1p")
fig.shift_origin(xshift="w+1c")
fig.plot(data=data, projection="X10c/10c", frame=True, region=[0, 10, 0, 10], pen="1p", close=True)
fig.shift_origin(xshift="w+1c")
fig.plot(data=data, projection="X10c/10c", frame=True, region=[0, 10, 0, 10], pen="1p", fill="lightblue")
fig.show()
```

Comment (seisman) [reply examples/gallery/maps/choropleth_map.py]: Here is an example showing how `close` works with/without fill.
```
import pygmt
fig = pygmt.Figure()
data = [[1, 1], [1, 5], [5, 5], [5, 1]]
fig.plot(data=data, projection="X10c/10c", frame=True, region=[0, 10, 0, 10], pen="1p")
fig.shift_origin(xshift="w+1c")
fig.plot(data=data, projection="X10c/10c", frame=True, region=[0, 10, 0, 10], pen="1p", close=True)
fig.shift_origin(xshift="w+1c")
fig.plot(data=data, projection="X10c/10c", frame=True, region=[0, 10, 0, 10], pen="1p", fill="lightblue")
fig.show()
```

File: examples/gallery/maps/choropleth_map.py [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+pygmt.makecpt(
+    cmap="acton",
+    series=[gdf["population"].min(), gdf["population"].max(), 10],
+    continuous=True,
+    reverse=True,
+)
+
+# Next, we plot the polygons and fill them using the defined
+# colormap. The target column is defined by the aspatial
+# parameter.
```
Comment (yvonnefroehlich) [inline examples/gallery/maps/choropleth_map.py:45]:
```suggestion
# Next, we plot the polygons and fill them using the defined colormap.
# The target column is defined by the aspatial parameter.
```

File: examples/gallery/maps/choropleth_map.py [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+fig.plot(
+    data=gdf,
+    pen="0.3p,gray10",
+    fill="+z",
+    cmap=True,
+    aspatial="Z=population",
+)
+
+
+# Add colorbar legend
```
Comment (yvonnefroehlich) [inline examples/gallery/maps/choropleth_map.py:55]:
```suggestion
    aspatial="Z=population",
)

# Add colorbar legend
```

**Comment by yvonnefroehlich** [comment]
Maybe we should make the PR title more specific and include the term "chorolepth map"?

File: examples/gallery/maps/choropleth_map.py (hunk 16 above)
Comment (yvonnefroehlich) [reply examples/gallery/maps/choropleth_map.py]: Thanks @seisman!

**Review by yvonnefroehlich — APPROVED** [review]
Beside the suggestions above, this example looks good to me!

File: examples/gallery/maps/choropleth_map.py (hunk 16 above)
Comment (yvonnefroehlich) [reply examples/gallery/maps/choropleth_map.py]: I am wondering whether we should update the [code example for the choropleth map of the EGU22 short course](https://www.generic-mapping-tools.org/egu22pygmt/ecosystem.html) and remove the `close` parameter for consistency, despite the issue that the choropleth map is not displayed correctly. Interestingly, I still do not face this issue when running this Jupyter notebook locally in JupyterLab, neither within a conda environment based on the provided yaml file nor within a conda environment with PyGMT dev version and GMT 6.4.0 installed.

**Pull request merged**